src/components/solution-control/solution-control-button/solution-control-button.tsx

```diff
@@ -2,60 +2,57 @@ import styles from './solution-control-button.module.css'
 
 import { ButtonHTMLAttributes, ReactElement } from 'react'
 import cn from 'classnames'
+import { observer } from 'mobx-react-lite'
 
 import { t } from '@i18n'
+import dtreeStore from '@store/dtree'
+import filterStore from '@store/filter'
 import { Icon } from '@ui/icon'
 
 interface ISolutionControlButtonProps
   extends ButtonHTMLAttributes<HTMLButtonElement> {
   solutionName: string | undefined
   controlName: string
   isOpen: boolean
-  isDeleteShown: boolean
-  onDeleteClick: () => void
+  isDeleteShown?: boolean
+  onDeleteClick?: () => void
 }
 
-export const SolutionControlButton = ({
-  className,
-  solutionName,
-  controlName,
-  isOpen,
-  isDeleteShown,
-  onDeleteClick,
-  ...buttonProps
-}: ISolutionControlButtonProps): ReactElement => {
-  return (
-    <button
-      className={cn(styles.solutionControlButton, className)}
-      {...buttonProps}
-    >
-      <span className={styles.solutionControlButton__label}>
-        {solutionName || t('solutionControl.selectSolution', { controlName })}
-      </span>
-      <span
-        className={cn(
-          styles.solutionControlButton__icon,
-          isDeleteShown && styles.solutionControlButton__icon_delete,
-        )}
+export const SolutionControlButton = observer(
+  ({
+    className,
+    solutionName,
+    controlName,
+    isOpen,
+    ...buttonProps
+  }: ISolutionControlButtonProps): ReactElement => {
+    const isModified =
```

Comment:
It's better to pass isModified as prop, rather than creating dependency on stores for generic component